src/main.cpp

```diff
@@ -2457,8 +2464,12 @@ static int64_t nTimeIndex = 0;
 static int64_t nTimeCallbacks = 0;
 static int64_t nTimeTotal = 0;
 
-bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pindex, CCoinsViewCache& view, bool fJustCheck)
+bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pindex, CCoinsViewCache& view, bool fJustCheck, bool fParallel)
```

Comment:
Taking bool as argument is generally recognized as an anti-pattern. See http://blog.iannelson.systems/back-to-basics-on-the-use-and-abuse-of-the-humble-boolean/ for instance.

I think this part of the code needs some refactoring before PV is introduced. For instance, extract the part of the block validation that are common between PV and regular validation in their own function, and have a base method which is different and handles the locking differently.